datafusion/core/tests/parquet/encryption.rs

```diff
@@ -284,7 +284,15 @@ fn verify_file_encrypted(
     options
         .options
         .insert("test_key".to_string(), "test value".to_string());
-    let object_path = object_store::path::Path::from(file_path.to_str().unwrap());
+
+    // the paths in encryption_factory are stored with "/", so we need to replace them on Windows
+    let file_path_str = if cfg!(target_os = "windows") {
+        file_path.to_str().unwrap().replace("\\", "/")
+    } else {
+        file_path.to_str().unwrap().to_owned()
+    };
```

Comment:
This confuses me a bit, specifically the comment as I don't think `encryption_factory` has anything to do with it? Is it possible to look for root cause of whats passing a Unix path to this function instead of patching here? 🤔

Reply:
I've tracked it down to when the Parquet files are written, where the path is transformed by the `ListingTableUrl::parse` function. Eventually those converted paths make their way to the encryption properties, resulting in the mismatch we see in the test. I do know if its a good idea to change that behavior, I think its better to change this particular test.